tests/system/Database/Live/ConnectTest.php

```diff
@@ -113,4 +119,74 @@ public function testConnectWithFailover()
 
         $this->assertGreaterThanOrEqual(0, count($db1->listTables()));
     }
+
+    /**
+     * @see https://github.com/codeigniter4/CodeIgniter4/issues/7550
+     */
+    public function testFinalDSNEstablishedDuringConnectionForPostgre()
+    {
+        if ($this->db->DBDriver !== 'Postgre') {
+            $this->markTestSkipped('Testing only on Postgre, as it has a specific implementation of DSN handling during connect.');
+        }
+
+        $this->expectException(ErrorException::class);
+
+        $db = Database::connect($this->group3);
+
+        try {
+            $db->connect();
+        } catch (ErrorException $e) {
+            $this->assertSame('Postgre', $this->getPrivateProperty($db, 'DBDriver'));
+
+            $expect = sprintf(
+                "host=%s port=5432 user=%s password='%s' dbname=%s",
+                $this->group3['hostname'],
+                $this->group3['username'],
+                $this->group3['password'],
+                $this->group3['database']
+            );
+            $this->assertSame($expect, $this->getPrivateProperty($db, 'DSN'));
+
+            throw $e;
+        }
+    }
+
+    /**
+     * @see https://github.com/codeigniter4/CodeIgniter4/issues/7550
+     */
+    public function testDSNStringEstablishedDuringConnectionForPostgre()
+    {
+        if ($this->db->DBDriver !== 'Postgre') {
+            $this->markTestSkipped('Testing only on Postgre, as it has a specific implementation of DSN handling during connect.');
+        }
+
+        $this->expectException(ErrorException::class);
+
+        $this->group3['DSN'] = sprintf(
+            'pgsql:host=%s;port=5432;user=%s;password=%s;dbname=%s',
+            $this->group3['hostname'],
+            $this->group3['username'],
+            $this->group3['password'],
+            $this->group3['database']
+        );
```

Comment:
Do we need to support the format `pgsql:host=%s;port=5432;user=%s;password=%s;dbname=%s`?
According to the docs, it is for PDO driver, but we don't have PDO drivers.
https://codeigniter4.github.io/CodeIgniter4/database/configuration.html#dsn

Reply:
Fair point. I'm not sure how commonly this DSN format is used right now.

I have nothing against depreciating it.

Reply:
I sent a PR to fix docs: https://github.com/codeigniter4/CodeIgniter4/pull/7555/files#diff-1bd8b31c324819bdbd0ce48dc5e41db938e9f9d6379d6c6e944055b7551ea362R38

PostgreSQL requires a DSN to connect, but CI4 builds it from other settings.
So we do not need to set `DSN` in Config File.